src/logic/basic.lean

```diff
@@ -1423,6 +1425,24 @@ lemma ite_eq_iff : ite P a b = c ↔ P ∧ a = c ∨ ¬ P ∧ b = c := by by_cas
 @[simp] lemma ite_eq_left_iff : ite P a b = a ↔ (¬ P → b = a) := by by_cases P; simp *
 @[simp] lemma ite_eq_right_iff : ite P a b = b ↔ (P → a = b) := by by_cases P; simp *
 
+@[simp] lemma ite_ne_left_iff : ite P a b ≠ a ↔ ¬ P ∧ a ≠ b :=
```

Comment:
It seems the error is caused in a situation where `simp` would have resolved the goal, but instead used one of these and ended up being unable to prove `a ≠ b`. Are we sure we want these lemmas as `simp`?

Reply:
Since the build is failing, I'll remove the awaiting-review tag.

Reply:
Looks like the simp set is a bit messed up, but I'm fine with unsimping them.